### DISCUSSION TITLE
Homekit translator

### DISCUSSION THREAD

**Discussion by bolfox** [Supergiovane/node-red-contrib-knx-ultimate#401] [Ideas]
Hi Max,

Here again! :-) Compliments for the latest updates; KNXUltimate is becoming more and more complete.

I like to integrate my knx setup with Apple Homekit with https://flows.nodered.org/node/node-red-contrib-homekit-bridged
The payload exiting from the HK service node (light bulb) is a JSON {"On":true}, which needs to be translated to a boolean.

I've used your HA translator, simply changing the input field to payload.On.

It possible to have a node with HK translation with HK icon? (Yes, i like to be be a perfectionist, sorry) also because the HA node isn't renominable (don't save the name field).

Also an ideal solution would be to have another node that translates the KNX output (boolean) and converts it into HA for status feedback.

Thank you for everything you do for us enthusiastic maniacs.

**Comment by Supergiovane** [comment]
The opposite node translation is a good idea, as well as icon. 
Will work on that monday!

**Comment by Supergiovane** [comment]
Just for info,
this will be delayed. I'm focused on KNX Secure now.
